src/finitio/support/heading.coffee

```diff
@@ -45,11 +46,21 @@ class Heading
   isEmpty: ->
     @size() == 0
 
-  allowExtra: ->
-    @options['allowExtra']
+  allowExtra: (value) ->
+    unless @options.allowExtra?
+      return false
+
+    unless value?
+      return true
+
+    extraType = @options.allowExtra
+    extraType.include(value)
```

Comment:
Keep `allowExtra` with a type/subtype semantics regarding the parameter. Add `allowExtraValue` instead.